xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +77,30 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
```

Comment:
You're writing a description of the public API here that's supposed to be useful for people trying to use the API.  Discussing that a method is used used by one particular piece of software using the public API isn't likely to be helpful here.

Also `items` is a purely internals concept.  Dropping "in items" would be better.

This reads oddly: "as well as updating the max_attained" - I'd either drop "the" or say something like "the max_attained value".

There's no documentation of what the parameters are meant to be, and being a template there's not even type information to give a clue.  All the user reading the docs can tell is that they're iterators but there's no clear indication of what sort of iterators or what over.

Also a sentence should end in a full stop.